### PR TITLE
Feature/species information #263

### PR DESCRIPTION
**Pull request by ray-millward-tessella**
Added the start of the species editing page along with redux state items.

It's not feature complete yet as I still need to add validation as well as handling images, but this is enough to start viewing species.

### CODE REVIEW

File: src/UI/state/speciesInformation/actions/upsertSpeciesInfo.action.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    }
+  }
+);
+
+export const getSpeciesInformation = createAsyncThunk(
+  'speciesInformation/getWithId',
+  async (id: string, { getState, dispatch }) => {
+    const token = (getState() as AppState).auth.token;
+
+    let res = await fetchGraphQlDataAuthenticated(
```
Comment (andykitch) [inline src/UI/state/speciesInformation/actions/upsertSpeciesInfo.action.ts:31]: Interesting that you've set this to use the authenticated endpoint - does it need to? I assume we'll reuse this action to get the species information for the information pages, which I don't think need to be auth-ed.

File: src/UI/state/speciesInformation/actions/upsertSpeciesInfo.action.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  'speciesInformation/upsert',
+  async (speciesInformation: SpeciesInformation, { getState }) => {
+    try {
+      const token = (getState() as AppState).auth.token;
+      await fetchGraphQlDataAuthenticated(
+        upsertSpeciesInformationMutation(speciesInformation),
+        token
+      );
+    } catch (e) {
+      console.log(e);
```
Comment (andykitch) [inline src/UI/state/speciesInformation/actions/upsertSpeciesInfo.action.ts:21]: Toast?

File: src/UI/components/speciesInformation/speciesInformationEditor.tsx [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+          sx={{ m: 0, minWidth: 150 }}
+        >
+          {id ? 'Update' : 'Create'}
+        </Button>
+      </div>
+    </div>
+  );
+};
+
+export default SpeciesInformationEditor;
```
Comment (andykitch) [inline src/UI/components/speciesInformation/speciesInformationEditor.tsx:109]: Ooh, no tests... Intentional?

File: src/UI/components/shared/textEditor/ToolbarPlugin.tsx [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+        },
+        LowPriority
+      )
+    );
+  }, [editor, updateToolbar]);
+
+  const handleBlockTypeChange = (e: SelectChangeEvent<string>) => {
+    setBlockType(e.target.value);
+
+    // if (e.target.value === 'h1') {
```
Comment (andykitch) [inline src/UI/components/shared/textEditor/ToolbarPlugin.tsx:99]: Commented out code?!

File: src/UI/state/speciesInformation/actions/upsertSpeciesInfo.action.ts (hunk 1 above)
Comment (ray-millward-tessella) [reply src/UI/state/speciesInformation/actions/upsertSpeciesInfo.action.ts:31]: You're quite right - this needs to be the unauthenticated route

File: src/UI/state/speciesInformation/actions/upsertSpeciesInfo.action.ts (hunk 2 above)
Comment (ray-millward-tessella) [reply src/UI/state/speciesInformation/actions/upsertSpeciesInfo.action.ts:21]: Yes, I'll extend this in the next PR to include notications for success as well.

File: src/UI/components/shared/textEditor/ToolbarPlugin.tsx (hunk 4 above)
Comment (ray-millward-tessella) [reply src/UI/components/shared/textEditor/ToolbarPlugin.tsx:99]: Yes I need to delete this.

File: src/UI/components/speciesInformation/speciesInformationEditor.tsx (hunk 3 above)
Comment (ray-millward-tessella) [reply src/UI/components/speciesInformation/speciesInformationEditor.tsx:109]: For the moment yes, in order to get Chris going on the next story - I'll come back and add tests for this one in my next PR.

File: src/UI/state/speciesInformation/actions/upsertSpeciesInfo.action.ts (hunk 1 above)
Comment (ray-millward-tessella) [reply src/UI/state/speciesInformation/actions/upsertSpeciesInfo.action.ts]: Fixed now

File: src/UI/components/shared/textEditor/ToolbarPlugin.tsx (hunk 4 above)
Comment (ray-millward-tessella) [reply src/UI/components/shared/textEditor/ToolbarPlugin.tsx]: Fixed.

**Review by andykitch — APPROVED** [review]

**Review by Christopher-Sandbach — APPROVED** [review]

**Pull request merged**